draft-ietf-webtrans-http2.md

```diff
@@ -269,6 +269,116 @@ exist, is not a stream established via extended CONNECT to use the
+connection error (Section 5.4.1) of type FRAME_SIZE_ERROR.
+
+## WT_STOP_SENDING Frame
+
+The RST_STREAM frame defined in HTTP/2 moves a stream to the "closed" state, but
+WebTransport streams can be reset unidirectionally, as in QUIC and HTTP/3.  A
+new HTTP/2 frame called WT_STOP_SENDING is introduced for an endpoint to
+unidirectionally reset a stream for reading.  WT_STOP_SENDING frames can be sent
+on a stream in the "open", or "half-closed (local)" state.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++---------------------------------------------------------------+
+|                        Error Code (32)                        |
++---------------------------------------------------------------+
+~~~
+{: #fig-wt_stop_sending title="WT_STOP_SENDING Frame Format"}
+
+The WT_STOP_SENDING frame contains a single unsigned, 32-bit integer identifying
+the error code (Section 7).  The error code indicates why the reading the stream
+is being abanonded.
+
+The WT_STOP_SENDING frame does not define any flags.
+
+The WT_STOP_SENDING frame half-closes the referenced stream and effects the same
+state machine transitions as sending or receiving the END_STREAM flag.  If the
+sender is in the open state, it transitions to "half-closed (remote)".  If the
+sender is in the "half-closed (local)" state, it transitions to "closed".  If
+the receiver is in the "open" state, it transitions to "half-closed (local)".
+If the receiver is in the "half-closed (remote)" state, it transitions to
+"closed".
+
+After receiving a WT_STOP_SENDING on a stream, the sender MUST NOT send
+additional frames for that stream.  After sending the WT_STOP_SENDING, the
+sending endpoint MUST be prepared to receive and handle additional frames sent
+on the stream that might have been sent by the peer prior to the arrival of the
+WT_STOP_SENDING.
+
+WT_STOP_SENDING frames MUST be associated with a WebTransport stream.  If a
+WT_STOP_SENDING frame is received with a stream identifier of 0x0, or a request
+or push stream, the recipient MUST treat this as a connection error of type
+PROTOCOL_ERROR.
+
+WT_STOP_SENDING frames MUST NOT be sent for a stream in any state other than
+"open" or "half-closed (local)".  It is possible to receive a WT_STOP_SENDING in
+another state however, because the sender might have closed or reset the stream
+while the WT_STOP_SENDING was in flight.  If a WT_STOP_SENDING frame identifying
+a stream that is already "closed" or "half-closed (local"), the recipient SHOULD
+ignore the frame.
+
+It is also possible to receive DATA frames on a WebTransport stream in the
+"half-closed (remote)" or "closed" states if the stream transitioned there via
+WT_STOP_SENDING.  If a DATA frame is received on a WebTransport stream in one of
+these states, the recipient MUST ignore the frame and credit the frame length to
```

Comment:
Is "credit" the right term here?  "count the frame length toward the FC window" perhaps.